pkg/drivers/kvm/kvm.go

```diff
@@ -443,16 +443,31 @@ func (d *Driver) Remove() error {
 	dom, err := conn.LookupDomainByName(d.MachineName)
 	if err != nil {
 		log.Warn("Domain %s does not exist, nothing to clean up...", d.MachineName)
+		return nil
 	}
-	if dom != nil {
-		log.Infof("Domain %s exists, removing...", d.MachineName)
-		if err := dom.Destroy(); err != nil {
-			return err
-		}
-		if err := dom.Undefine(); err != nil {
-			return err
-		}
+
+	log.Infof("Domain %s exists, removing...", d.MachineName)
+	if err := d.destroyRunningDomain(dom); err != nil {
+		return errors.Wrap(err, "destroying running domain")
+	}
+
+	if err := dom.Undefine(); err != nil {
```

Comment:
Since this function doesn't do anything after handling this error, just use "return dom.Undefine()".

Reply:
hey, I haven't changed this because on the other upcoming PR I'm changing this code, should I still change it nonetheless?

Reply:
ignore, it makes sense to change, so the commit itself is 100% no matter which code came later.